### PR TITLE
drop docker as a dependency in kubetest

### LINKED ISSUE #19667: building broken due to bad dependency
**What happened**:

```
go: downloading github.com/moby/buildkit v0.7.2
go: extracting github.com/moby/buildkit v0.7.2
go: github.com/moby/buildkit@v0.7.2 requires
	github.com/containerd/containerd@v1.4.0-0: reading https://proxy.golang.org/github.com/containerd/containerd/@v/v1.4.0-0.mod: 410 Gone
gazelle: finding module path for import github.com/moby/buildkit/client: exit status 1: go: github.com/moby/buildkit@v0.7.2 requires
	github.com/containerd/containerd@v1.4.0-0: reading https://proxy.golang.org/github.com/containerd/containerd/@v/v1.4.0-0.mod: 410 Gone
gazelle: finding module path for import github.com/moby/buildkit/session: exit status 1: go: github.com/moby/buildkit@v0.7.2 requires
	github.com/containerd/containerd@v1.4.0-0: reading https://proxy.golang.org/github.com/containerd/containerd/@v/v1.4.0-0.mod: 410 Gone
gazelle: finding module path for import github.com/moby/buildkit/session: exit status 1: go: github.com/moby/buildkit@v0.7.2 requires
	github.com/containerd/containerd@v1.4.0-0: reading https://proxy.golang.org/github.com/containerd/containerd/@v/v1.4.0-0.mod: 410 Gone
```

https://storage.googleapis.com/kubernetes-jenkins/pr-logs/pull/test-infra/19586/pull-test-infra-bazel/1319387076549939200/build-log.txt

https://proxy.golang.org/github.com/containerd/containerd/@v/v1.4.0-0.mod
>  not found: github.com/containerd/containerd@v1.4.0-0: invalid version: unknown revision v1.4.0-0

**What you expected to happen**:

builds should succeed

**How to reproduce it (as minimally and precisely as possible)**:

run pull-test-infra-bazel

**Please provide links to example occurrences, if any**:

https://storage.googleapis.com/kubernetes-jenkins/pr-logs/pull/test-infra/19586/pull-test-infra-bazel/1319387076549939200/build-log.txt

**Anything else we need to know?**:

#sig-testing discussion from here https://kubernetes.slack.com/archives/C09QZ4DQB/p1603401820355500

### PR DESCRIPTION
**Pull request by BenTheElder**
I _think_ this helps with #19667

### CODE REVIEW

**Comment by BenTheElder** [comment]
/hold
still getting pulled in somehow

File: go.mod [hunk 1]
```diff
@@ -11,7 +11,6 @@ replace github.com/golang/lint => golang.org/x/lint v0.0.0-20190301231843-5614ed
 replace (
 	cloud.google.com/go/pubsub => cloud.google.com/go/pubsub v1.3.1
 	github.com/Azure/go-autorest => github.com/Azure/go-autorest v12.2.0+incompatible
-	github.com/docker/docker => github.com/docker/docker v1.4.2-0.20200203170920-46ec8731fbce
```
Comment (BenTheElder) [inline go.mod:14]: /hold
I suppose we may still need this then, I'm not sure what's pulling it in currently.

**Comment by BenTheElder** [comment]
https://github.com/kubernetes-sigs/controller-runtime/commit/17eebbff0d4860a005997ca22c4b29a07019448d#diff-3295df7234525439d778f1b282d146a4f1ff6b415248aaac074e8042d9f42d63

not present in a non-alpha release yet (v0.7.0-alpha.4  v0.7.0-alpha.3 v0.7.0-alpha.2 v0.7.0-alpha.1 v0.7.0-alpha.0)

File: go.mod (hunk 1 above)
Comment (BenTheElder) [reply go.mod:14]: `go mod vendor && grep -R docker vendor/` doesn't show any imports, maybe a workspace dependency ...

File: kubetest/local.go [hunk 2]
```diff
@@ -27,57 +26,13 @@ import (
...
-
-	// reverse sort by Creation time so we delete newest containers first
-	sort.Slice(containers, func(i, j int) bool {
-		return containers[i].Created > containers[j].Created
-	})
-
-	// stop then remove (which implicitly kills) each container
-	duration := time.Second * 1
-	removeOptions := types.ContainerRemoveOptions{
-		RemoveVolumes: true,
```
Comment (amwat) [inline kubetest/local.go:61]: would we also need `-v` in the `docker rm` replacement to maintain the same behavior ?

Comment (BenTheElder) [reply kubetest/local.go:61]: Good catch!

**Review by amwat — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**